### PR TITLE
Birthday feature

### PR DESCRIPTION
**Pull request by CindyTsai1**
Added birthday parser
Added birthday attribute (on-going)
Added birthday prefix
Updated command parser for birthday attribute
Added birthday list (on-going)

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -14,6 +15,7 @@
 import seedu.address.logic.commands.AddCommand;
 import seedu.address.logic.parser.exceptions.ParseException;
 import seedu.address.model.person.Address;
+import seedu.address.model.person.Birthday;
```
Comment (April0616) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:21]: merge master and then continue :) or need to solve tons of conflicts %>_<%

File: src/main/java/seedu/address/model/person/Birthday.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class Birthday {
+
+    public static final String MESSAGE_BIRTHDAY_CONSTRAINTS =
+            "Person's birthday should be in the format of DDMMYYYY, and it should not be blank";
+
+    /*
+     * The first character of the address must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String BIRTHDAY_VALIDATION_REGEX = "\\d{2}\\d{2}\\d{4}";
```
Comment (April0616) [inline src/main/java/seedu/address/model/person/Birthday.java:20]: I think \\d{8} equals to \\d{2}\\d{2}\\d{4}?

And... do we need to check DDMMYYYY, DD is in the range 01 - 31, MM is in the range 01-12, YYYY is in the range 1900 - 201x? today
I think maybe VALIDATION_REGEX=([0-3])([0-9])([0,1])([0-9])([1-2])(//d{3}),
what do you think? :)

File: src/main/java/seedu/address/model/person/Birthday.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            throw new IllegalValueException(MESSAGE_BIRTHDAY_CONSTRAINTS);
+        }
+        this.date = birthday;
+    }
+
+    /**
+     * Returns true if a given string is a valid person birthday.
+     */
+    public static boolean isValidBirthday(String test) {
+        return test.matches(BIRTHDAY_VALIDATION_REGEX);
```
Comment (April0616) [inline src/main/java/seedu/address/model/person/Birthday.java:42]: or further check birthday here?

File: src/test/java/seedu/address/model/person/BirthdayTest.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    @Test
+    public void isValidBirthday() {
+        // invalid birthdays
+        assertFalse(Birthday.isValidBirthday("")); // empty string
+        assertFalse(Birthday.isValidBirthday(" ")); // spaces only
+        assertFalse(Birthday.isValidBirthday("1324")); // less than 8 digits
+        assertFalse(Birthday.isValidBirthday("dgd")); // non-numeric
+        assertFalse(Birthday.isValidBirthday("23/05/1997")); // invalid slash
+        assertFalse(Birthday.isValidBirthday("23 05 1997")); // invalid space
+
```
Comment (April0616) [inline src/test/java/seedu/address/model/person/BirthdayTest.java:19]: maybe add more tests
assertFalse(Birthday.isValidBirthday("00000000");
assertFalse(Birthday.isValidBirthday("99661998");
:)

**Review by April0616 — DISMISSED** [review]
Generally good code!!
Just a little advice!!

**Review by nbriannl — APPROVED** [review]
Consider working on yuting's requests for v1.3. The feature is workable in it's current state. 👍

**Pull request merged**